pkg/mirror-producer/producer_test.go

```diff
@@ -0,0 +1,888 @@
+// Copyright 2018 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	cnitypes "github.com/containernetworking/cni/pkg/types"
+	types040 "github.com/containernetworking/cni/pkg/types/040"
+	current "github.com/containernetworking/cni/pkg/types/100"
+	cniversion "github.com/containernetworking/cni/pkg/version"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/containernetworking/plugins/pkg/testutils"
+	"github.com/vishvananda/netlink"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	plugin "github.com/k8snetworkplumbingwg/ovs-cni/pkg/plugin"
+
+	"github.com/k8snetworkplumbingwg/ovs-cni/pkg/types"
+)
+
+type MirrorNet040 struct {
```

Comment:
Please add a comment explaining why we treat 0.4.0 separatelly

Reply:
You are right, it'a good idea.
Also plugin_test.go doesn't have a comment about this (Net040 and NetCurrent).
We should write a comment and add it to both plugin, mirror_producer and mirror_consumer files.
Do you have a suggestion? The difference is about `PrevResult` type, but how can we explain this in a clear way?

Reply:
It can be as simple as the explanation here I think: https://github.com/k8snetworkplumbingwg/ovs-cni/pull/227/files#diff-14c657ea5956b13b208b921905d0d631a1835176df2a42bd4a3f1bb42bdae160R160

add prevResult from cache, because missing in CNI spec < 0.4.0

Reply:
Actually now when I look at it again, I don't think we need that at all... I was confused before thinking that this is related to prevresult being related from cache.

Reply:
Ok, should I remove the comment here
```
// MirrorNet040 struct that represent the network configuration for CNI spec 0.4.0
type MirrorNet040 struct {
	...
}

// MirrorNetCurrent struct that represent the network configuration for CNI spec 1.0.0
type MirrorNetCurrent struct {
	...
}
```

or can I leave it?

Reply:
Let's leave it as is :)